### PR TITLE
Functions to write things explicitly to files

### PR DESCRIPTION
**Pull request by Victor-C-Zhang**
Functions to write and re-read `Graph`, `Supernode`, and `Sketch` objects to secondary storage.

### CODE REVIEW

File: graph.cpp [hunk 1]
```diff
@@ -27,13 +27,29 @@ Graph::Graph(uint64_t num_nodes): num_nodes(num_nodes) {
   GraphWorker::startWorkers(this, bf);
 }
 
+Graph::Graph(std::ifstream& file) {
```
Comment (etwest) [inline graph.cpp:30]: So the way we're thinking about using this in multiple CC is to reset an existing graph's sketches using the data on disk. How would we accomplish that using this setup?

File: graph.cpp [hunk 2]
```diff
@@ -27,13 +27,29 @@ Graph::Graph(uint64_t num_nodes): num_nodes(num_nodes) {
...
+#endif
+  representatives = new set<Node>();
+  supernodes = new Supernode*[num_nodes];
+  parent = new Node[num_nodes];
+  for (Node i=0;i<num_nodes;++i) {
+    representatives->insert(i);
+    supernodes[i] = new Supernode(num_nodes, file);
+    parent[i] = i;
+  }
+  is_copy = true;
```
Comment (etwest) [inline graph.cpp:43]: Same with this variable. There are a number of places where we do different things for copies. How would that fit into what we want to do with this.

File: graph.cpp (hunk 1 above)
Comment (Victor-C-Zhang) [reply graph.cpp:30]: Constructing from a filestream saves only the supernode and metadata fields. It wasn't made with the intention of being able to be a functional ingestion machine after construction. That's why I use the term "reheated" like you would yesterday's dinner. I'll rethink this whole process in light of our new need for a snapshot-based serialization.

**Comment by Victor-C-Zhang** [comment]
Save this for later

**Pull request closed without merging**